src/lib/OS/SimOS.cc

```diff
@@ -315,10 +308,10 @@ int64_t SimOS::cloneProcess(uint64_t flags, uint64_t stackPtr,
   newProc->context_.regFile[retReg.type][retReg.tag] = {0, 8};
   // Update stack pointer
   newProc->context_.sp = stackPtr;
-  if (Config::get()["Core"]["ISA"].as<std::string>() == "rv64") {
+  if (SimInfo::getISA() == ISA::RV64) {
     newProc->context_.regFile[arch::riscv::RegisterType::GENERAL][2] = {
         stackPtr, 8};
-  } else if (Config::get()["Core"]["ISA"].as<std::string>() == "AArch64") {
+  } else if (SimInfo::getISA() == ISA::AArch64) {
     newProc->context_.regFile[arch::aarch64::RegisterType::GENERAL][31] = {
         stackPtr, 8};
     // Set appropriate system register to TLS value.
```

Comment:
I cannot comment on the exact line (the 2 below this comment) but updating the TPIDR_EL0 sysreg should be done in a similar way to the DCZ sysreg. Also, the ARM64_SYSREG_TAGS enum can probably be removed in favour of the simInfo implementation

Reply:
Changed logic and retired ARM64_SYSREG_TAGS